src/components/TooltipNew/PopperTooltip.js

```diff
@@ -33,8 +37,13 @@ class PopperTooltip extends React.PureComponent {
       document.addEventListener('click', this.handleDocumentClick);
     }
 
+    if (this.props.closeWithEsc && this.getIsVisible()) {
+      document.addEventListener('keydown', this.handleDocumentKeyDown);
+    }
+
     if (didHide) {
       document.removeEventListener('click', this.handleDocumentClick);
+      document.removeEventListener('keydown', this.handleDocumentKeyDown);
```

Comment:
remove event listener also in `componentWillUnmount`

Reply:
right